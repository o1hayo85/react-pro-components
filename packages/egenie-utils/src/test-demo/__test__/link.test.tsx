import React from 'react';
import renderer from 'react-test-renderer';
import { Link } from '../link';

test('Link', () => {
  let component: renderer.ReactTestRenderer;
  renderer.act(() => {
    component = renderer.create(
      <Link page="http://www.facebook.com">
        Facebook
      </Link>
    );
  });

  let tree = component.toJSON() as renderer.ReactTestRendererJSON;
  expect(tree).toMatchSnapshot();

  renderer.act(() => {
    // manually trigger the callback
    tree.props.onMouseEnter();
  });

  // re-rendering
  tree = component.toJSON() as renderer.ReactTestRendererJSON;
  expect(tree).toMatchSnapshot();

  renderer.act(() => {
    // manually trigger the callback
    tree.props.onMouseLeave();
  });

  // re-rendering
  tree = component.toJSON() as renderer.ReactTestRendererJSON;
  expect(tree).toMatchSnapshot();
});
