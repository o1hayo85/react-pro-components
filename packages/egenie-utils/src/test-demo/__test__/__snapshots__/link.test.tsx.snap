// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Link 1`] = `
<DocumentFragment>
  <a
    class="normal"
    href="http://www.facebook.com"
  >
    Facebook
  </a>
</DocumentFragment>
`;

exports[`Link 2`] = `
<DocumentFragment>
  <a
    class="hovered"
    href="http://www.facebook.com"
  >
    Facebook
  </a>
</DocumentFragment>
`;

exports[`Link 3`] = `
<DocumentFragment>
  <a
    class="normal"
    href="http://www.facebook.com"
  >
    Facebook
  </a>
</DocumentFragment>
`;
